### PR TITLE
UID2 module: Add documentation for cstg

### PR DESCRIPTION
**Pull request by jingyi-gao-ttd**
## Type of change
- [X] Other

## Description of change
Update documentation for Client-Side Token Generation feature

## Other information
PR for this feature: https://github.com/prebid/Prebid.js/pull/10510

### CODE REVIEW

File: modules/uid2IdSystem.md [hunk 1]
```diff
@@ -1,11 +1,46 @@
...
+
+```
+    $ gulp build --modules=uid2IdSystem --disable UID2_CSTG
+```
+
 **Important information:** UID2 is not designed to be used where GDPR applies. The module checks the passed-in consent data and will not operate if the `gdprApplies` flag is true.
 
+## Client-Side Token Generation (CSTG) mode
+
+For publishers seeking a purely client-side integration without the complexities of server-side involvement, the CSTG mode is highly recommended. This mode necessitates the provision of a public key, subscription ID, and directly identifying information (DII). In the CSTG mode, the module takes on the responsibility of encrypting the DII, generating the UID2 token, and handling token refreshes when necessary.
```
Comment (sunnywu) [inline modules/uid2IdSystem.md:17]: _This mode necessitates the provision of a public key, subscription ID, and directly identifying information (DII)_

Maybe change to:

_This mode requires the provision of a public key, subscription ID and directly identifying information (DII) - either emails or phone numbers._

File: modules/uid2IdSystem.md [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+
+To configure the module to use this mode, you must:
+1. Set `parmas.serverPublicKey`  and `params.subscription` 
+2. Provide **ONLY ONE DII** by set `params.email`/`params.phone`/`params.emailHash`/`params.phoneHash`
```
Comment (sunnywu) [inline modules/uid2IdSystem.md:21]: _2. Provide **ONLY ONE DII** by setting ONLY ONE of `params.email`/`params.phone`/`params.emailHash`/`params.phoneHash`_

?

File: modules/uid2IdSystem.md [hunk 3]
```diff
@@ -1,11 +1,46 @@
...
+
```
Comment (sunnywu) [inline modules/uid2IdSystem.md:22]: Add a new sentence after 2. :

_For the correct way to generate email and phone hash, refer to this [UID2 documentation](https://unifiedid.com/docs/getting-started/gs-normalization-encoding)_

?

File: modules/uid2IdSystem.md [hunk 4]
```diff
@@ -1,11 +1,46 @@
...
+               email: 'user@email.com',
+               //phone: '+0000000',
+               //emailHash: '...email hash...',
+               //phoneHash: '...phone hash ...'
+            }
+        }]
+    }
+});
+```
+
```
Comment (sunnywu) [inline modules/uid2IdSystem.md:43]: Do we need to add the description of the new CSTG specific params in the 
_Parameter Descriptions for the usersync Configuration Section_

at the bottom of the page?

Comment (jingyi-gao-ttd) [reply modules/uid2IdSystem.md:45]: I've addressed the changes and added CSTG specific params to the parameter description table

Comment (sunnywu) [reply modules/uid2IdSystem.md:45]: Thanks let's wait for Kimberley to come back to confirm the CSTG feature naming and then we can further modify this PR if necessary

File: modules/uid2IdSystem.md [hunk 5]
```diff
@@ -1,11 +1,48 @@
...
 
+## Client-Side Token Generation (CSTG) mode
+
+For publishers seeking a purely client-side integration without the complexities of server-side involvement, the CSTG mode is highly recommended. This mode requires the provision of a public key, subscription ID and [directly identifying information (DII)](https://unifiedid.com/docs/ref-info/glossary-uid#gl-dii) - either emails or phone numbers. In the CSTG mode, the module takes on the responsibility of encrypting the DII, generating the UID2 token, and handling token refreshes when necessary.
+
+To configure the module to use this mode, you must:
+1. Set `parmas.serverPublicKey`  and `params.subscriptionId` 
+2. Provide **ONLY ONE DII** by setting **ONLY ONE** of `params.email`/`params.phone`/`params.emailHash`/`params.phoneHash`
+
+For the correct way to generate email and phone hash, refer to this [UID2 documentation](https://unifiedid.com/docs/getting-started/gs-normalization-encoding).
```
Comment (sunnywu) [inline modules/uid2IdSystem.md:23]: Matt has just now created [this table ](https://atlassian.thetradedesk.com/confluence/display/UID2/UID2+CSTG+Integration+Instructions#UID2CSTGIntegrationInstructions-Generateatoken) earlier. Maybe you can tweak it and add to here to tell pubs clearly they may need to do normalisation/hashing depending on the param input type?

You can still tweak and add these lines?
_Note that setIdentityFromEmail will normalize email addresses, but setIdentityFromPhone requires phone numbers to be normalized.

Refer to this page [Normalization and Encoding](https://unifiedid.com/docs/getting-started/gs-normalization-encoding) on [unifiedid.com](http://unifiedid.com/) for details on email address normalization, SHA-256 hashing and Base64 encoding._

File: modules/uid2IdSystem.md (hunk 4 above)
Comment (sunnywu) [reply modules/uid2IdSystem.md:45]: Still a while before you formally raise this PR to prebid, but before you submit it later, please ask Gen our tech writer to review your changes first tho

File: modules/uid2IdSystem.md (hunk 5 above)
Comment (sunnywu) [reply modules/uid2IdSystem.md]: Thanks @jingyi-gao-ttd latest changes looks good!

File: modules/uid2IdSystem.md [hunk 6]
```diff
@@ -1,11 +1,60 @@
...
+|------------------|-------------------------------------------------------|--------------------------------------|--------------------------------|
+| params.email     | When you have users' email address                    | No                                   | No                             |
+| params.phone     | When you have user's phone number                     | Yes                                  | No                             |
+| params.emailHash | When you have user's hashed, normalized email address | Yes                                  | Yes                            |
+| params.phoneHash | When you have user's hashed, normalized phone number  | Yes                                  | Yes                            |
+
+
+*Note that setIdentityFromEmail will normalize email addresses, but setIdentityFromPhone requires phone numbers to be normalized.*
+
+Refer to this page [Normalization and Encoding](https://unifiedid.com/docs/getting-started/gs-normalization-encoding) on [unifiedid.com](https://unifiedid.com/) for details on email address normalization, SHA-256 hashing and Base64 encoding.
```
Comment (patmmccann) [inline modules/uid2IdSystem.md:35]: I'm nervous this link might break, is it possible to replicate here?

File: modules/uid2IdSystem.md [hunk 7]
```diff
@@ -1,11 +1,61 @@
...
 
 *Server Only* mode was originally referred to as *legacy mode*, but it is a popular mode for new integrations where publishers prefer to handle token refresh server-side.
 
+*Client-Side Token Generation* mode is included in UID2 module by default. However, it's important to note that this mode is currently in an alpha stage of development. For publishers who do not intend to use it, you have the option to instruct the build to exclude the code related to this feature:
+
+```
+    $ gulp build --modules=uid2IdSystem --disable UID2_CSTG
+```
+If you do plan to use Client-Side Token Generation (CSTG) mode, we strongly recommend consulting with the UID2 Team before implementing it in your integration.
```
Comment (sunnywu) [inline modules/uid2IdSystem.md:12]: Maybe change to:

_If you do plan to use Client-Side Token Generation (CSTG) mode, please consult the UID2 Team first as they will provide required configuration values for you to use (see the Client-Side Token Generation (CSTG) mode section below for details)_

File: modules/uid2IdSystem.md (hunk 4 above)
Comment (sunnywu) [reply modules/uid2IdSystem.md:58]: let's push ahead with current naming until PMs can decide on the final naming

File: modules/uid2IdSystem.md [hunk 8]
```diff
@@ -1,11 +1,61 @@
...
+
+```
+    $ gulp build --modules=uid2IdSystem --disable UID2_CSTG
+```
+If you do plan to use Client-Side Token Generation (CSTG) mode, we strongly recommend consulting with the UID2 Team before implementing it in your integration. 
+
 **Important information:** UID2 is not designed to be used where GDPR applies. The module checks the passed-in consent data and will not operate if the `gdprApplies` flag is true.
 
+## Client-Side Token Generation (CSTG) mode
+
```
Comment (sunnywu) [inline modules/uid2IdSystem.md:17]: Maybe add a note in bold:

**_This mode is currently in the alpha stage of development. Please consult UID2 Team first as they will provide required configuration values for you to use._**

File: modules/uid2IdSystem.md (hunk 6 above)
Comment (jingyi-gao-ttd) [reply modules/uid2IdSystem.md]: I've put that to the end

**Review by sunnywu — APPROVED** [review]

**Review by patmmccann — APPROVED** [review]

**Pull request merged**